contracts/RoundManager.sol

```diff
@@ -79,8 +79,42 @@ contract RoundManager is Ownable, ProviderPool, ProviderManager {
     }
   }
 
-  // Getter function for ActiveProviderSet
+  function removeActiveProvider(address _provider) internal {
+    ActiveProviderSet storage aps = activeProviderSets[roundNumber];
+    require(aps.isActive[_provider]);
+    // Remove from providers[];
+    //   - Find index of _provider in providers array
+    uint length = aps.providers.length;
+    uint i = 0;
+    while (aps.providers[i] != _provider) {
```

Comment:
This is brutal, is there no way to index this better?

Reply:
I guess what we could do is not remove from the array, and say that a Provider is currently active if he is in aps.providers AND if aps.isActive(_provider) is true

Reply:
And to find out whether they resigned or not (ie still active), you'd have to check `aps.isActive`